src/utils/auth.ts

```diff
@@ -176,27 +183,42 @@ export default class Auth {
           },
           (error) => {
             console.error(error)
-            this.logout()
-            reject(error)
+            this.logout().then(() => reject(error))
           },
         )
       }
     })
   }
 
   clearLocalStorage() {
-    if (typeof localStorage !== 'undefined') {
-      localStorage.removeItem(ACCESS_TOKEN_KEY)
-      localStorage.removeItem(EXPIRES_AT_KEY)
-      localStorage.removeItem(USER_KEY)
-      localStorage.removeItem(VIEWING_AS_USER_KEY)
-    }
+    return new Promise((resolve) => {
+      const removeLocalStorage = async () => {
+        cookie.remove(ACCESS_TOKEN_KEY, {
+          domain: process.env.NEXT_PUBLIC_AUTH_COOKIE_DOMAIN,
```

Comment:
this one is a big **gotcha** because it has to have the domain or it doesn't actually clear it :trollface:

Reply:
I think the way this cookie was being removed async was causing heartache and doing it this way is better.

Reply:
I thought it was a cool use of a serverless function, but yeah, it sure seemed like it was at least in part to blame for the timing issues.